test/extended/dr/static_pod_rollouts.go

```diff
@@ -0,0 +1,106 @@
+	"k8s.io/apimachinery/pkg/types"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Suite:openshift/etcd/recovery][Disruptive] etcd", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("etcd-static-pod-rollouts").AsAdmin()
+
+	etcdNamespace := "openshift-etcd"
+	var initialEtcdPodCount int   // the number of etcd pods in the initial state of the cluster
+	var etcdTargetNodeName string // node from which the etcd pod is to be removed
+
+	g.BeforeEach(func() {
+		isSingleNode, err := exutil.IsSingleNode(context.Background(), oc.AdminConfigClient())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSingleNode {
+			g.Skip("the test is for etcd peer communication which is not valid for single node")
+		}
+
+		etcdPods, err := e2epod.GetPods(context.Background(), oc.AdminKubeClient(), etcdNamespace, map[string]string{"app": "etcd"})
+		o.Expect(err).ToNot(o.HaveOccurred())
+		initialEtcdPodCount = len(etcdPods)
+
+		masterNodes := masterNodes(oc)
+		etcdTargetNodeName = masterNodes[0].Name
+	})
+
+	g.AfterEach(func(ctx context.Context) {
+
+		// Reset the hardware speed back to default to leave the cluster in the same state.
+		g.GinkgoT().Log("resetting hardware speed back to default")
+		data := fmt.Sprintf(`{"spec": {"controlPlaneHardwareSpeed": ""}}`)
+		_, err := oc.AdminOperatorClient().OperatorV1().Etcds().Patch(ctx, "cluster", types.MergePatchType, []byte(data), metav1.PatchOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("debugging into the same node to add back the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("-n", etcdNamespace, "node/"+etcdTargetNodeName, "--", "chroot", "/host", "/bin/bash", "-c", "mv /var/lib/etcd-backup/etcd-pod.yaml /etc/kubernetes/manifests/").Execute()
+		o.Expect(err).ToNot(o.HaveOccurred(), "failed to add etcd static pod back to the node")
+
+		g.GinkgoT().Log("waiting for all the etcd instances to be available to return the cluster to its original state")
+		waitForReadyEtcdStaticPods(oc.AdminKubeClient(), initialEtcdPodCount)
+
+	})
+
+	g.It("is able to block the rollout of a revision when the quorum is not safe", func(ctx context.Context) {
+		var err error
+
+		g.GinkgoT().Log("debugging into the node to remove the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("-n", etcdNamespace, "node/"+etcdTargetNodeName, "--", "chroot", "/host", "/bin/bash", "-c", "mkdir -p /var/lib/etcd-backup && mv /etc/kubernetes/manifests/etcd-pod.yaml /var/lib/etcd-backup").Execute()
+		err = errors.Wrap(err, "failed to remove etcd static pod from the node")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("waiting for the etcd pod to be removed")
+		waitForReadyEtcdStaticPods(oc.AdminKubeClient(), initialEtcdPodCount-1)
+
+		// allow time for the operator status to be updated after the removal of etcd pod
+		time.Sleep(20 * time.Second)
+
```

Comment:
There is a delay between the removal of `etcd pod` and the updation of controller statuses in `etcd/cluster`. This was causing the test to fail. Hence added a small delay here.

Reply:
please avoid `time.Sleep`, the environments are too different for it to guess the timing. Please use wait.poll to await a specific condition you're looking for.

Reply:
Replaced the sleep with wait.poll in the mentioned line. 

There is a sleep (here)[https://github.com/openshift/origin/blob/17bf685572b7757e34d93833478e8c02accefafa/test/extended/dr/static_pod_rollouts.go#L102] after triggering a rollout to observe if the state of the system has changed. Couldn't think of a way to avoid using sleep here, as for a successful test we aren't expecting any change in condition. @tjungblu Do you have any recommendations ?